content/en/docs/guides/image-update.md

```diff
@@ -627,396 +627,99 @@ flux create image policy podinfo \
+     path: /spec/template/spec/containers/0/args/-
+     value: --aws-autologin-for-ecr
+```
+
+#### Using Native GCP GCR Auto-Login
+
+There is native support for the GCP Google Container Registry available since `image-reflector-controller` [vadd-version](link to changelog)
+which was released with Flux release v[add-flux-ver]. This feature is enabled by setting the `--gcp-autologin-for-gcr`
+flag. This works with both clusters that have Workload Identity enabled, and those that use the default service account.
+This flag can be added by including a patch in the `kustomization.yaml` overlay file in your `flux-system`,
+similar to the process described in [customize Flux manifests](../installation.md/#customize-flux-manifests):
+                                             
 ```yaml
-spec:
-  secretRef:
-    name: gcr-credentials
-```
-
-#### Using a JSON key [long-lived]
-
-{{% alert color="info" title="Less secure option" color="warning" %}}
-From [Google documentation on authenticating container registry](https://cloud.google.com/container-registry/docs/advanced-authentication#json-key)
-> A user-managed key-pair that you can use as a credential for a service account.
-> Because the credential is long-lived, it is the least secure option of all the available authentication methods.
-> When possible, use an access token or another available authentication method to reduce the risk of
-> unauthorized access to your artifacts. If you must use a service account key,
-> ensure that you follow best practices for managing credentials.
-{{% /alert %}}
-
-A Json key doesn't expire, so we don't need a cronjob,
-we just need to create the secret and reference it in the ImagePolicy.
-
-First, create a json key file by following this
-[documentation](https://cloud.google.com/container-registry/docs/advanced-authentication).
-Grant the service account the role of `Container Registry Service Agent`
-so that it can access GCR and download the json file.
-
-Then create a secret, encrypt it using [Mozilla SOPS](mozilla-sops.md)
-or [Sealed Secrets](sealed-secrets.md) , commit and push the encypted file to git.
-
-```sh
-kubectl create secret docker-registry <secret-name> \
-  --docker-server=<GCR-REGISTRY> \ # e.g gcr.io
-  --docker-username=_json_key \
-  --docker-password="$(cat <downloaded-json-file>)"
-```
-
-### Azure Container Registry
+patches:
+- target:
+    version: v1
+    group: apps
+    kind: Deployment
+    name: image-reflector-controller
+    namespace: flux-system
+  patch: |-
+    - op: add
+      path: /spec/template/spec/containers/0/args/-
+      value: --gcp-autologin-for-gcr
+ ### add this patch to annotate service account if you are using Workload identity
+- target:
+    version: v1
+    group: ""
+    kind: ServiceAccount
+    name: image-reflector-controller
+    namespace: flux-system
+  patch: |-
+    - op: add
+      path: /metadata/annotations/spec/
+      value: <gcp-service-account-name>.<PROJECT_ID>.iam.gserviceaccount.com
```

Comment:
```suggestion
- |-
  apiVersion: v1
  kind: ServiceAccount
  metadata:
    name: image-reflector-controller
    namespace: flux-system
    annotations:
      iam.gke.io/gcp-service-account: <gcp-service-account-name>@<PROJECT_ID>.iam.gserviceaccount.com
```

Reply:
Note the `@` replacement

I tested several incorrect formations of the `patchesStrategicMerge` before landing on this one 🙄 please double check this is correct, I believe I have it now.